### PR TITLE
Shuttle nav compuer docking cameras can no longer swap across z-levels.

### PR DESCRIPTION
**Pull request by Timberpoes**
## About The Pull Request

Shuttle docking nav computers can pick locations across z-levels. This is most likely an oversight, as shuttles should only be able to dock using this console on their own z-level.

Currently doesn't do anything bad as Icebox is the only map that supports this sort of multi-Z camera movement and has a distinct lack of pilotable shuttles, but patching this before it becomes an issue.

## Why It's Good For The Game

Oversight feex good.

## Changelog
:cl:
fix: Shuttle navigation computers can no longer switch to z-levels the shuttle is not present on.
/:cl:

### CODE REVIEW

**Comment by Dennok** [comment]
Why you remove MultiZ support? 
Now shuttles cant be propertly used on multiz maps, but later.

**Comment by Timberpoes** [comment]
You use the shuttle on Multi-Z maps by going to that Z-level with the shuttle itself first, or by placing a beacon on the Z-level that the nav computer can lock on to.

Navigation computers are not intended to allow shuttles **to dock on z-levels they're not even on and potentially can't even fly to**.

I'm not even going to attempt to deal with the hornet's nest that are nav consoles being able to scout z-levels with their Eye of Meson. At the very least installing a nav gigabeacon on the z-level should be the pre-requisite to it.

This doesn't mean shuttles can't be used across Z-levels. It means that their nav computers still need a beacon on every z-level they want to custom dock at.

**Comment by Dennok** [comment]
You think about this levels like ordinary space levels connected by map magic, but on multiZ map levels connected by multiZ magic to work like one .
Why you can't move docking camera if human can move by ladder? 


About meson view. What if make docking cameras view do not distinguish between turfs types. See only like able/unable to land?

**Comment by Timberpoes** [comment]
> 
> 
> I'm curious in what situation a shuttle wouldn't be able to go up or down. Or why you'd need a beacon at that height to travel to it.

tl;dr - The default behaviour should be to assume something can't dock at a z-level unless there's something that explicitly says it should be able to. This doesn't prevent ships docking at nearby Z-levels, it just prevents them doing it by default without something explicitly saying they can.

If I take Icebox as an example - If we ever added a nav-computer compatible shuttle of some sort to it, why can it go to the lower Z-levels? They're supposed to be caverns. Ships really shouldn't be flying down at all, at least a nav gigabeacon would allow them to do a mega ultra bluespace jump through using bluespace magic to get there.

Similar to KS13, Lavaland and the station's primary Z-level, if you want a shuttle to be able to go to lower Z-levels you can do it at the mapping level by adding the appropriate zones that the Nav Console can pick up on. KS13 has this exact functionality, along with the station Z-level and Lavaland. Or the players can do it themselves by constructing the nav gigabeacons.

This way, the map designers have some level of control over whether a ship can **natively** fly to and/or dock at a given z-level depending on how much sense it makes in the context of that map and that z-level.

**Comment by Jared-Fogle** [comment]
By z-level, do you just mean depth? Or can shuttles still fly to different *space* z-levels? (Or could they never do that?)

**Comment by Dennok** [comment]
> By z-level, do you just mean depth? Or can shuttles no longer fly to different _space_ z-levels? (Or could they never do that?)

It remove ability to docker eye move vertically in multiZ maps

**Comment by Timberpoes** [comment]
I probably didn't explain clearly enough in the PR body.

Right now we require some sort of beacon or mapping element to be on a z-level to enable a shuttle to use the docking camera there.

https://github.com/tgstation/tgstation/pull/54043 added actions for advanced camera consoles to move up and down z-levels. 

This behaviour was inherited by nav consoles (thanks OOP), allowing their docking camera to flick between Z-levels in the same way an advanced camera could without any checks for if it SHOULD be able to flick between Z-levels in the same scenarios. I would classify this behaviour as an oversight, as it allows remote viewing and docking at Z-levels a ship may not actually be able to fly directly to.

The default behaviour for nav computers should be, in my opinion, to by default disallow custom docking at a "near" Z-level (as in, higher or lower station z-levels) unless something explicitly allows it on the Z-level, at which point the same interface that works for "far" z-levels takes over.

**Comment by Dennok** [comment]
#54043 added actions for advanced camera consoles to move up and down in MULTI Z MAPS, not in standard maps.

Why we cant move eye where humans can standard move (not transit or teleport.), it can fall , and move up with jetpack?
Our multi z simulates direct connect of z lvls with 3.5m height.

**Comment by Timberpoes** [comment]
> 
> 
> #54043 added actions for advanced camera consoles to move up and down in MULTI Z MAPS, not in standard maps.
> 
> Why we cant move eye where humans can standard move (not transit or teleport.), it can fall , and move up with jetpack?
> Our multi z simulates direct connect of z lvls with 3.5m height.

The Nav Computer is **nothing** to do with CCTV networks or the station's camera net, despite inheriting from the advanced camera console.

Once again, I do not want this to be the default behaviour of Nav Computers. Making behaviour like this default is how coding or mapping oversights cause the kinds of bugs that can be abused instead of the kinds of bugs that can't be abused.

Once again, let us use the example of Icebox - Just because you can flick between the CCTV networks of each floor does not mean that your shuttle's nav computer has vision of every single floor. Just because you have a shuttle with vision on the surface of Icebox does not mean that your shuttle AUTOMATICALLY has vision of the **caverns and caves** underground that form part of the planet's z-levels.

If a given Z-level is intended to be Nav Computer dockable, then either players in-round or mappers should put something on that Z-level to allow it. We should not wholesale allow Nav Computers to view entire Z-levels by default. This behaviour works for advanced cams consoles because if they flick to a z-level with no cameras, all they see is static.On nav computers, they see a meson-vision cross section of the entire Z-level **AND** they can dock on that Z-level. Once again, this should not be the DEFAULT behaviour. It should be enabled by mappers placing the approrpiate elements of through the actions of the players after the round has started.
